src/exchange/exchange_client.rs

```diff
@@ -91,11 +93,34 @@ impl ExchangeClient {
             info.meta().await?
         };
 
+        let spot_meta = if let Some(spot_meta) = spot_meta {
+            spot_meta
+        } else {
+            let info = InfoClient::new(None, Some(base_url)).await?;
+            info.spot_meta().await?
+        };
+
         let mut coin_to_asset = HashMap::new();
         for (asset_ind, asset) in meta.universe.iter().enumerate() {
             coin_to_asset.insert(asset.name.clone(), asset_ind as u32);
         }
 
+        let parse_spot_name_to_u32 = |name: &str| {
+            name.replacen('@', "", 1).parse::<u32>().unwrap()
+        };
+
+        for (ind, asset) in spot_meta.universe.iter().enumerate() {
```

Comment:
There is no need to enumerate, also you don't use spot_meta after this loop so you can consume it with into_iter to avoid cloning the name

Reply:
Not enumerating anymore